sklearn/utils/estimator_checks.py

```diff
@@ -112,7 +112,11 @@ def _yield_checks(estimator):
             # We skip pairwise because the data is not pairwise
             yield check_sample_weights_shape
             yield check_sample_weights_not_overwritten
-            yield check_sample_weight_equivalence
+            for sparse_container in [None] + CSR_CONTAINERS:
```

Comment:
I would rather not test for all possible sparse containers but only a single one, e.g. `csr_array`. The point of this estimator check is not to check that the models behave correctly with all kinds of sparse data, but just to make sure that if the model has a sparse specific code-path for some solvers, then the sample weight properties still hold in those branches of the code.

We don't want to generate redundant estimator checks because they are expensive to run given the number of estimators in scikit-learn.